File: content/courses/parallel-computing-introduction/codes/omp_outdep.py
import numpy as np
from omp4py import *

@omp
def output_dependency(N):
    a=np.zeros(N,dtype='int')
    x=5;
    a[N-1]=100;

    with omp("parallel for"):
        for i in range(N-1):
            a[i]=i
            a[i+1]=x+i

    return a

N=48;
a=output_dependency(N)
for i in range(N):
    print(i,a[i])

File: content/courses/parallel-computing-introduction/codes/omp_outdep.c
#include <stdio.h>
#include <omp.h>

int main() {

    const int N=48;
    int a[N];
    int x=5;

    a[N-1]=100;

#pragma omp parallel for 
    for (int i=0; i<N-1; i++) {
        a[i]=i;
        a[i+1]=x+i;
    }

    for (int i=0; i<N; i++) {
        printf("%d %d\n",i,a[i]);
    }

}
